Update templates

a2line opened the pull request:
#### Update templates
* `perso`:
   * Old modules configuration system removed.
* `individu`:
   * New buttons to navigate to previous/next Sosa number when browsing with a Sosa reference.
   * Add a link to implex·es graph·s between the parents on consanginity %.
* `updfam`: 
   * Hide useless inputs for spouses and children with javascript when select=link.
   * Add navigation bar with scrollspy, revamp validation buttons (request on issue #548).
* `cousins_tools`(new):
   * Buttons to show/hide `spouse` and `image`, ± buttons for `v1` and `v2` now allow every possible degrees of kindship.
* `destable`/`ancsosa`:
   * Add internal navigation in tables in 1st column. `num`, new button “only selected generation”.
* `ancsosa`:
   * Added option `show table by couple` (table of ascendants of templm).
   * Added CSV mode to export tables.
* `deslist_hr` (new in defaut template):
   * Tree view of descendants of templm to “replace” d’Aboville.
* `destable`, `deslist_hr`:
   * Added options to identify and reduce implexes.
* `buttons` (new):
   * Horizontal navigation submenu bar between trees/tree views/tables of ascendants/descendants concentrating all previously added options.
* Add five personnal modules in tree section:
  * Family tree (a5 and `m=F`).
  * H-tree 7 gen. (a6).
  * Mixed tree 9 gen. (a7).
  * HI-tree (a8).
  * Descendant tree (a9).
* `menubar`: updated with all needed, better adding support of all new file/option, lot of minor changesmove some direct buttons to `menubar` becaus shortcuts are coming for them (see #583).
* Add a evar option `wide` included in `$prefix;` to switch between container and container-fluid classes.
* Use a monospace font on the select used to switch languages.
* Few fixes: js on forms, "VD" month typo (#517).
* Fix grammar for fratrie module/lexicon (#568/#569).
* Add exponant function for calculation in templates, syntax is `%expr(x^y)`
* Add translations for Russian (thanks @yaroslav-bogatov)

#### Update CSS/JS
* Update jquery.min.js to version 3.3.1
* Update Bootstrap from v.4-alpha-6 to release 4.1.1
  * Update Boostrap js/css (Bootstrap CDN fixed to release)
  * Replace theter.js with popper.min.js version 1.14.3 from https://popper.js.org/ > included in bootstrap.bundle.js
* Update Font Awesome to version 5.1
* Update autosize.js to version 4.0.2

(Updated 2018-06-21)

hgouraud committed:
%lang; to replace %if;(bvar.default_lang='')%if;(evar.lang='')...

hgouraud committed:
Implement the %lang; command in all templates

hgouraud committed:
Wrong [on %s's side:::] translation in english (fix #568)

a2line commented:
Push forced to fix last problems (see todo of 1st post).

I have added the modifications of Dominique PR #555 on this one (minus the changes on forms and modules individu/arbre_horizontal).

Any remarks?

dominique95880 commented:
Bonjour

Je reviens un instant sur l'affaire des mois exprimés directement en calendrier républicain.
En effet la liste des mois autorisés est en MAJUSCULE (VD|...|JC) et le mode normal du clavier est d'être en minuscule.
Je pense qu'il faudrait "allonger" la liste des valeurs autorisées (VD|...|JC) en incluant les mois en minuscule (vd|...|jc)

Par exemple   %let;month_input;pattern="(?:0?[1-9]|1[0-2]|VD|BR|FM|NI|PL|VT|GE|FL|PR|ME|TH|FT|JC|vd|br|fm|ni|pl|vt|ge|fl|pr|me|th|ft|jc)" size="1" maxlength="2" xcond%in;

Je vous laisse le soin de mettre en place cette modification ou de trouver une autre astuce pour faire en sorte que vd etc. soient admis au même titre que VD ETC.

Dominique

a2line commented:
Merci @dominique95880 pour cette remarque. J'ai fait un commit en ce sens.

GuillaumeBrochu commented:
Perhaps could you include this PR (or something equivalent...) about the correct display of sources & notes of burial vs cremation for your big template update?
https://github.com/geneweb/geneweb/pull/559
https://github.com/geneweb/geneweb/pull/559/files

hgouraud committed:
Add links to famille, 7gen, 9gen and HI trees in menubar

hgouraud committed:
m=A;t=T;t1=7/h7/M/CT accordingg to cases. Direct access in ancestors drop down menu

hgouraud committed:
Fixed the unbound variable 0; issue

a2line committed:
fix nearly all personinfo

hgouraud committed:
Inversion: navigate on sosa nbr and open person on picture

hgouraud committed:
Suppress nav=, remove extre

hgouraud committed:
Some style adjustments. Buttons for arbre_7h

hgouraud committed:
Removed outline in various buttons

hgouraud committed:
Removed outline in arbre_vertical buttons

hgouraud committed:
Test for sosa 1 in anctree_hi to avoid undefined ancestor.spouse.is_male...

hgouraud committed:
record commit number and Ocaml versions in distribution folder

hgouraud committed:
Fixed ancestor.spouse.is_male issue

hgouraud committed:
Fixed Undefined marrriage_date issue

hgouraud committed:
Fixed the Failed: expected int error in arbre_horizontal

hgouraud committed:
Undefined variable in arbre_h7, error in perso.txt (evar.test)

hgouraud committed:
Removed button outline in chronology

Cleanup of anctree_hi

Revert to xxx='on' coding style. Add (occ) in title where missing

remode prefix_no_lang

hgouraud committed:
prefix_no_lang

hgouraud committed:
Missing spaces between parameters in anctree_hi

hgouraud committed:
clearfix in anctree.txt and arbre_9gen

hgouraud committed:
Replace <br clear='all'> by <div style='clear:both;'></div>

a2line committed:
show a dark disabled bouton on evar.v and evar.v

a2line committed:
update popper.min.js to 1.12.6, update bootstrap.js/css to v4-dev

a2line committed:
use %prefix_no_lang; to avoid spaces in URLs from pz or nz

pz/nz have sometimes spaces (error in w3c) and %sosa_ref.first_/surname_key; are not available everywhere so we use new %prefix_no_lang; to maintain same sosa_ref in URL
evar.oc/ocz should still be hidden if empty or null and not added to URL

a2line committed:
add missing doctype and lang, cleaning code

a2line committed:
fix double sosa line if self=sosa_ref, use &#010; instead of &#013 for line breaks

hgouraud committed:
Fixed home/back/wide/search buttons on menubar.txt (+ stray ';' in url)

a2line committed:
add complete date and place on titles, fix sosa order if missing, add last missing personinfo

hgouraud committed:
Coded evar.cgl into anctree_hi.txt

hgouraud committed:
Proposition pour supprimer de prefix et suffix les evar vides et oc/ocz=0 (quid iz, nz, pz?)

a2line committed:
missing add family buttons if sex is neuter

hgouraud committed:
Cleanup of several modules (missing cancel_links tests)

hgouraud committed:
Install tree modification, à la templm, in default template

hgouraud committed:
Allow for dag=on/off in A trees

hgouraud committed:
Small translate correction in ancmenu

hgouraud committed:
Fine tuning upddag.txt. Add home.txt with home/wide/search/back buttons

hgouraud committed:
Suppress rebuild_url in dag.txt, arbre_vertical.txt, anctree.txt upddag.txt

hgouraud committed:
some more removal of rebuild_url, ancsosa, anctree, destable, arbres_9/7/hi/v, anctree

hgouraud committed:
Rebuild_url is gone

hgouraud committed:
%prefix_no_wide; not needed anymore

hgouraud committed:
Small bug in notes.txt (message Mick on GeneWeb ML)

a2line committed:
import home in menubar

a2line committed:
fix m/t/v env.var positions in URLs

hgouraud committed:
Suppress browsing_with_sosa in home.txt which triggers undefined variable in gwd.log

hgouraud committed:
Fix image=on in arbre_famille. Move new key to end of url

hgouraud committed:
Make image=on or image= default value. Url contains image=off or nothing

hgouraud committed:
Add (occ) if occ!='' in title of page

hgouraud committed:
New template based version of descendant tree (op_m=a9 and menubar).

hgouraud committed:
Marriage date on same line as &

hgouraud committed:
Better symbol for spouses. Title for person and  brothers and sisters navigation

hgouraud committed:
Do not display & year if evar.spouse!=on and no children

hgouraud committed:
In arbre_famille buttons, replace & by fa-user-plus fa-user

hgouraud committed:
Add marriage count to &, marriage date and spouse =on at start

hgouraud committed:
Extraneous ( in unions.txt

a2line committed:
brute update to Font Awesome v5, not loading brand icons

a2line committed:
update Bootstrap 4 to beta 3 including popper.js v1.12.9

a2line committed:
fix switch from desc to anc table, keep same order for options

a2line committed:
fix fa5 icons in menubar

hgouraud committed:
Change fa-photo into fa-image

hgouraud committed:
Inversion d'un titre (implex) dans arbre_hi_tools

a2line committed:
fix radio buttons, group insert input/buttons

hgouraud committed:
In MOD_IND, reorder sex choice into H ? F

a2line committed:
update Bootstrap v4.0.0 final

a2line committed:
add autofocus on anchor for mod_data_ok

a2line commented:
Bootstrap is now in version 4.0.0 (no more alpha/beta…). This big PR still needs a few fixes for icons (switch to Font Awesome 5) and to be cleaned before being merged!!

hgouraud committed:
Correct ancmenu.txt (two superfluous buttons)

hgouraud committed:
Upgrade a few icons to awesome 5 in menubar

a2line committed:
remove big and unused atm .svg font files

a2line committed:
update jquery.min.js to 3.3.1

fablhx commented:
Should this be merged ?

a2line committed:
Use image=off, fix icons

a2line committed:
more compact calendar, better buttons

a2line commented:
I have rebased this PR on current master (lots of conflicts with destable/ancsosa/menubar…) and cherry-picked #566 commits (`%lang;` variable) to clean it (77 modified files because of a bad regexp, for 38 really modified ones).

We need a few days to check if nothing has been broken before merging (this last rebase was painfull!) and perhaps also to test it on Tuxfamily for CGI mode (new js/font files).

a2line committed:
missing num=on; for ancestor table

a2line committed:
Add full dates in tooltips, same links for ancestors table

hgouraud committed:
Env.key oc and ocz were discarded in suffix and prefix_no_all

a2line committed:
use env_binding when switching lang, revert br for tables

a2line committed:
Fix W3C warnings on tables, remove duplicate ids on spouses

hgouraud committed:
Scan notes and sources do replace quotes and apostrophes by their &#xx; value

a2line committed:
Add a navigation button on num cell (not complete)

hgouraud committed:
Add %svar similar to %pvar and %qvar but for sosa numbers

a2line committed:
Finish navigation button on tables, env.key never move in URLs, shorten sources with less CR

a2line committed:
Disable destable link if no child, print generations only up to max_desc_level

a2line committed:
Restore %access; for oc=, 3 light yellow color for hover in tables

a2line committed:
fix various forms and w3c error, revamp welcome a bit

a2line committed:
remove extra line breaking templm bindings

a2line committed:
update bs

a2line committed:
update FontAwesome to 5.0.8, fix CGI removing ../ in urls

a2line committed:
keep %suffix in URLs only if digest is empty

a2line committed:
remove text-muted for blue buttons

a2line committed:
allow longer labels for sex/occ inputs

a2line committed:
templm don't need jquery twice

a2line committed:
remove BOM from file

a2line committed:
better login input in cgi, wider sosa_ref and motd

a2line committed:
uses slashs in titles instead of commas, fix missing closing tag

a2line committed:
temporary replace %prefix_base; to keep login in URL in CGI

a2line committed:
navigate to spouse.access when t=I

a2line committed:
allow sosa/cgl menu for guests and friends

a2line committed:
less typo and w3c warnings, fix u2 images when cgl=on

a2line committed:
move %hidden var inside form

a2line committed:
verifiy spouse=event.spouse to prevent event.date spreading

a2line committed:
add nbsp before units in string_of_age

hgouraud committed:
Redundant sosa=on in menubar.txt

a2line committed:
temporary fix for CGI

hgouraud committed:
New %prefix_base_passwd; function (WIP)

a2line committed:
update jquery.line.js to trunk

a2line committed:
Use prefix_base_password instead of prefix_base except in index

a2line committed:
prevent iz stacking from multiple lang selection

hgouraud committed:
Missing on_side in templm/perso.txt and cleanup

hgouraud committed:
Removed extra space and . in union.txt for separation

hgouraud committed:
Some more adjusts in unions.txt (spaces and one ,)

hgouraud committed:
Suppress class-ml-1 in link to wedding-notes

hgouraud committed:
Adjust spaces and , in children weddings

a2line committed:
ADD_PAR link in red if not ancestor.has_parents

a2line committed:
add child name in h1 (ADD_PAR), fix div to hide

hgouraud committed:
Removed perso_module_tmplnb in perso.txt. Default is p_mod=zz

a2line committed:
missing evar.num in ancmenu, fix cgl on num

hgouraud committed:
remove padding and margins in person display for destree.txt

hgouraud committed:
Remove <br> in display_person in destree.txt

a2line committed:
a few more relationships

a2line committed:
fix typo ggrruncles

hgouraud committed:
Slight programming style adjustment (and color change) in anctree_hi.txt

hgouraud committed:
(Util.transl conf :) in cousins.ml (probably applicable elsewhere as well)

hgouraud committed:
Apply (Util.transl conf ':') in remaining files (a few error messages remain)

a2line committed:
add more relationships

hgouraud committed:
correct a stupid bug preventing compilation of %tvar;

hgouraud committed:
Display intermediate parent data in cousins lists, navigate with +-
buttons

hgouraud committed:
In chronologie, set death.date.year to 9999 if alive

hgouraud committed:
prefix_2, prefix_base_2, prefix_base_password_2 without trailing &

a2line committed:
fix suffix in copyr

a2line committed:
extend the option “only the selected generation” to destable, new buttons for this option

a2line committed:
distinctive bibliography header before sources

a2line committed:
update bootstrap to nearly 4.1 release

a2line committed:
revert column design in perso, better contrasts for colors in css1

hgouraud committed:
A1D1, A0Dn, as template solution (pass 1)

a2line committed:
fix encoding of .htm files in gwsetup

everything is iso-8859-1 (Latin-1) except intro.txt.dos

hgouraud committed:
Some more cousins for A0Dn and AnD0 (wip)

a2line committed:
update autosize.js to 4.0.1

a2line committed:
update Font Awesome to 5.0.9

hgouraud committed:
Invert up-to/only generation, specify generation in h1 title in only

a2line committed:
add scrollspy to forms nav

hgouraud committed:
Proposal to correct capitale_utf_8

hgouraud committed:
Treat also ÿ and C4, C5, C6, C7 cases

a2line committed:
move submit button to 1st card

a2line committed:
missing union notes

a2line committed:
remove max on occ inputs, ? ? can be very high on big databases

a2line committed:
update Bootstrap to v4.1.0

a2line committed:
fix icon for sosa, permute age/title

a2line committed:
fix parse_error2 in gwd log, add access in menubar, try other form for -+ buttons

a2line committed:
fix js to show other_evts

a2line committed:
show relation tree if cgl=on and count>0 not 1

hgouraud committed:
In cousins.ml, count spouses and display total

a2line committed:
use focus reference on image/delete/chg_evt

a2line committed:
better child count using uppercase

hgouraud committed:
Fix hr in updfam parent card. Ok button at bottom. Test for nb_parents needed

hgouraud committed:
Add NoSexesCheckMarried to the test for marriage in the list of 20 most recent

hgouraud committed:
Correct chronologie to show event witness

hgouraud committed:
Slight change in notesLink.ml to allow for [[fn/sn/oc!=0]]. Temporary hack

hgouraud committed:
Remove unused Wiki_info triggering a warning

GuillaumeBrochu commented:
Je viens d'essayer cette branche bs4beta. Le rendu est vraiment beau, et les menus sont vraiment pratiques et bien pensés.

Beau travail!

hgouraud commented:
Merci pour le maximum de tests sur cette branche!
On est ouverts aux suggestions et améliorations.
Ceci dit, on se concentre sur le nettoyage pour préparer la v7 beta!!

Henri

hgouraud committed:
Missing %apply;capitalize in <h1> title of welcome.txt

hgouraud committed:
Put back capitale for Stammbaum in german

a2line committed:
try other fonts in temlpm

hgouraud committed:
Adjust anclist.txt to replace implex subtree by 'see above' in horizontal trees

a2line committed:
temporary removal of unfinished image mode in descendant section

a2line committed:
remove bottom-margin under sources h2

a2line committed:
scrollspy only on defaut temnplate

a2line committed:
remove useless %/ in tags

a2line committed:
clearer delete mode, missing num=on on desc. table

a2line committed:
use box drawing characters, use monospace on table, shorten anchor

a2line committed:
reorder datalist loading

a2line committed:
remove duplicate entry

hgouraud committed:
(^) arrow to ancestor.same, with anchor. Button to control display of implex in horizontal tree

hgouraud committed:
Bring Mickael Roue's deslist into default world ad d'Aboville tree

a2line committed:
temporary +1 fix for l_v because internal %max_desc_level; is sometimes wrong

hgouraud committed:
Options buttons for deslist_mr

a2line committed:
fix URLs of buttons in d’Aboville tree, better accessibility for templm

hgouraud committed:
Some workaround max_desc_level problem. Better count of persons

hgouraud committed:
Static_max_desc_level for better control of + button

hgouraud committed:
deslist_mr -> deslist_hr. ; replaced by &, before variable

a2line committed:
use & in URLs of buttons, accessibility design for templm

a2line committed:
new template file for buttons for easier maintenance

hgouraud committed:
bring menubar and buttons to anclist.txt

hgouraud committed:
Merge see-above in bs4beta. Buttons for deslist_hr

hgouraud committed:
Include horizontal tree in menubar. Some cleanup in destable and ancsosa

hgouraud committed:
Highlight target when hovering on the implex symbol

hgouraud committed:
Highlight on mouseover for multi occurences and implex in deslist_hr.txt

a2line committed:
add em=R link on ancsosa == implexes

hgouraud committed:
Hilight and jump to label hidden under letter and ┳ symbon

hgouraud committed:
deslist_hr now suppresses implex if requested

hgouraud committed:
l211 and l212 were incorrectly computed

hgouraud committed:
Better link to spouse , case of multiple marriages

hgouraud committed:
Reduction of implex applied to destable. No visible indication yet

hgouraud committed:
Effort to report implex in destable with popover.

a2line committed:
fix en/fr implex/implexes

hgouraud committed:
Fix count en count1 in calls to table_row

hgouraud committed:
Remove popover in d'Aboville for implex

hgouraud committed:
Test for cancel_links for implex indication

a2line committed:
reintroduce skip-navigation for screenreaders

a2line committed:
update autosize to 4.02

a2line committed:
copy asc. table by couple to defaut template

a2line committed:
fix w3c errors/warnings

hgouraud committed:
Suppress extra buttons in anclist, use perso_utils for togen

hgouraud committed:
%sp correction in ancsosa. Access to ancsosa (m=A;t=Z) with templm

a2line committed:
fix image size and vertical align of male td

hgouraud committed:
Better logic for implex reduction

hgouraud committed:
Better logic for implex in deslist_hr

hgouraud committed:
is_%index;_%family.index; helps solve multiple marriage issues

a2line committed:
update Bootstrap to 4.1.1

a2line committed:
update popper to 1.14.3

a2line committed:
restore missing box-drawing characters by fixing cgl=on

hgouraud committed:
Missing if(cancel_links) in anclist

hgouraud committed:
Some unbound variables in cousins_tools and arbre_descendant

hgouraud committed:
Simplify %tvar.i.s to %tvar.s as i is implicitly sosa_ref.index

hgouraud committed:
Slight improve to new WiKi link syntax (fn/sn/0 was badly treated)

hgouraud committed:
Replace include.trl by base_trailer

hgouraud committed:
add navigation buttons: previous/next ancestor of sosa_ref

hgouraud committed:
%sosa.lvl, %tvar.sosa; in indiividu.txt

hgouraud committed:
%tvar; becoems %sosa_anc;

hgouraud committed:
Add expr(n^p)

hgouraud committed:
Correct self!=sosa_ref into self.index!=sosa_ref.index (was failing on ancestors with same fn sn)

hgouraud committed:
Limit sosa navigation to max_anc_level of sosa_ref

hgouraud committed:
Solution to move up in sosa nav in spite of implex. No sol for down

a2line committed:
allow navigation with accesses by key, add min/max to limit a bit more

hgouraud committed:
Cover x^0 = 1

hgouraud committed:
New functions %sosa_next; and %sosa_prev; allowing full sosa navigation, up and down

a2line committed:
remove relationship with sosa_ref if sosa_ref

a2line committed:
add relationship between parents on consanguinity %

hgouraud committed:
%sosa_anc_p; to find ancestors of the current person

hgouraud committed:
Adjust Sosa.gen to be coherent with %sosa.lvl;

a2line committed:
disable marriage button on dag

a2line committed:
add copy wiki link buttons to menu bar

a2line committed:
update Font Awesome to v5.0.13

a2line committed:
add sosa navigation to left aligned menubar, use new icons

a2line committed:
missing scrollspy for ADD_PAR

a2line committed:
fix sosa nav stuck on unknown ind (? ?) by excludind env.key i=

a2line committed:
copy link to clipboard works from everywhere using data- attribute of buttons

mix first solutions given in https://stackoverflow.com/questions/400212/how-do-i-copy-to-the-clipboard-in-javascript

a2line committed:
move wide button to p_mod dropdown

a2line committed:
new option and button to show the table of ancestors in CSV format

a2line committed:
fix before/after date precision, add %title; to CSV

yaroslav-bogatov committed:
add translations for russian

cyrillic letters in UTF-8 are multibyte, delete 2 bytes to subsitute one letter

a2line committed:
bootstrap.bundle.js includes popper.js, use recommanded stackpath CDN

a2line committed:
restore toolbar and accent with more characters, shorten parents card

hgouraud committed:
Suppress <p> ... </p> in comments

a2line committed:
actions done on family uses ip not i

a2line committed:
user safer variables, “ind” can be contained in first names or surnames

a2line committed:
add comma before title in destree cells

a2line committed:
fix typo in titles

a2line committed:
better floating buttons, fix translation

a2line committed:
update Font Awesome to v5.1

a2line committed:
move css and css files to subdirs

a2line committed:
rename perso_module to modules, move images of modules to subdir

a2line commented:
:heart_eyes:
